src/pages/workspace/accounting/qbo/export/QuickbooksOutOfPocketExpenseEntitySelectPage.tsx

```diff
@@ -14,44 +17,63 @@ import * as Policy from '@userActions/Policy';
 import CONST from '@src/CONST';
 import ROUTES from '@src/ROUTES';
 
+type CardListItem = ListItem & {
+    value: ValueOf<typeof CONST.QUICKBOOKS_EXPORT_ENTITY>;
+    isShown: boolean;
+};
+type CardsSection = SectionListData<CardListItem, Section<CardListItem>>;
+
 function QuickbooksOutOfPocketExpenseEntitySelectPage({policy}: WithPolicyProps) {
     const {translate} = useLocalize();
     const styles = useThemeStyles();
-    const {exportEntity, syncTaxes} = policy?.connections?.quickbooksOnline?.config ?? {};
+    const {exportEntity, syncTaxes, syncLocations} = policy?.connections?.quickbooksOnline?.config ?? {};
+    const isLocationsEnabled = Boolean(syncLocations && syncLocations !== CONST.INTEGRATION_ENTITY_MAP_TYPES.NONE);
     const isTaxesEnabled = Boolean(syncTaxes && syncTaxes !== CONST.INTEGRATION_ENTITY_MAP_TYPES.NONE);
-
+    const isTaxError = isTaxesEnabled && exportEntity === CONST.QUICKBOOKS_EXPORT_ENTITY.JOURNAL_ENTRY;
+    const isLocationError = isLocationsEnabled && exportEntity !== CONST.QUICKBOOKS_EXPORT_ENTITY.JOURNAL_ENTRY;
     const policyID = policy?.id ?? '';
-    const data = [
-        {
-            value: CONST.QUICKBOOKS_EXPORT_ENTITY.CHECK,
-            text: translate(`workspace.qbo.check`),
-            keyForList: CONST.QUICKBOOKS_EXPORT_ENTITY.CHECK,
-            isSelected: exportEntity === CONST.QUICKBOOKS_EXPORT_ENTITY.CHECK,
-            isShown: true,
-        },
-        {
-            value: CONST.QUICKBOOKS_EXPORT_ENTITY.JOURNAL_ENTRY,
-            text: translate(`workspace.qbo.journalEntry`),
-            keyForList: CONST.QUICKBOOKS_EXPORT_ENTITY.JOURNAL_ENTRY,
-            isSelected: exportEntity === CONST.QUICKBOOKS_EXPORT_ENTITY.JOURNAL_ENTRY,
-            isShown: !isTaxesEnabled,
-        },
-        {
-            value: CONST.QUICKBOOKS_EXPORT_ENTITY.VENDOR_BILL,
-            text: translate(`workspace.qbo.vendorBill`),
-            keyForList: CONST.QUICKBOOKS_EXPORT_ENTITY.VENDOR_BILL,
-            isSelected: exportEntity === CONST.QUICKBOOKS_EXPORT_ENTITY.VENDOR_BILL,
-            isShown: true,
-        },
-    ];
+
+    useEffect(() => {
+        if (!isTaxError && !isLocationError) {
+            return;
+        }
+        Policy.updatePolicyConnectionConfig(policyID, CONST.QUICK_BOOKS_CONFIG.EXPORT_ENTITY, '');
+    }, [policyID, isTaxError, isLocationError]);
```

Comment:
Can you please explain the reason behind this?

Reply:
@s77rt this is an example:
<img width="692" alt="Screenshot 2024-04-18 at 11 31 02" src="https://github.com/Expensify/App/assets/28352309/decf3752-8e21-4b2b-a8a7-e97c3b527302">
we need to clean up value on the right screen (screenshot) when value is not available - but was selected before

Reply:
On the left screen above, shouldn't there be a label above "Journal entry"? I think the mock is bad, it should say:

Export as
Journal entry

Like all other push rows that have a value.

Reply:
@shawnborton yeah - mock was not correct - but i added that logic during implementation:
<img width="374" alt="Screenshot 2024-04-18 at 13 11 00" src="https://github.com/Expensify/App/assets/28352309/7daae5b8-0d40-4bf2-a17a-a545a1243b18">

Reply:
Nice. The hint text feels a little far away from the row above it though, can you make the hint text closer to it?

Reply:
@shawnborton how about this:
<img width="375" alt="Screenshot 2024-04-18 at 13 31 26" src="https://github.com/Expensify/App/assets/28352309/8f8f3120-2868-4551-b9a6-56a297c0bd14">